src/Plugin/Condition/TemplateWhisperer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\template_whisperer\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\template_whisperer\TemplateWhispererManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Template Whisperer' condition.
+ *
+ * @Condition(
+ *   id = "template_whisperer",
+ *   label = @Translation("Template Whisperer"),
+ *   context = {
+ *     "node" = @ContextDefinition("entity:node", label = @Translation("Node"))
+ *   }
+ * )
+ */
+class TemplateWhisperer extends ConditionPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Template Manager.
+   *
+   * @var \Drupal\template_whisperer\TemplateWhispererManager
+   */
+  protected $templateWhispererManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityStorageInterface $entity_storage, array $configuration, $plugin_id, $plugin_definition, TemplateWhispererManager $template_whisperer_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->templateWhispererManager = $template_whisperer_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $container->get('entity.manager')->getStorage('node_type'),
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('plugin.manager.template_whisperer')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['#attached']['library'][] = 'template_whisperer/block';
+
+    $options = $this->templateWhispererManager->getList();
+
+    $form['suggestions'] = [
+      '#type' => 'checkboxes',
+      '#title' => $this->t('When the node has the following Suggestion(s)'),
+      '#default_value' => $this->configuration['suggestions'],
+      '#options' => array_map('\Drupal\Component\Utility\Html::escape', $options),
+      '#description' => $this->t('Select suggestion(s) to enforce only on those selected. If none are selected, all suggestion will be allowed.'),
+    ];
+
+    $form = parent::buildConfigurationForm($form, $form_state);
+    unset($form['negate']);
```

Comment:
By unsetting this, you will export this config on `node` even when nothing is checked (it cause an error with drupal form submission which need this field) & cause an unattended issues that will trigger this condition to every non-node bundle (aka taxonomy-term) to "not show" the given block (because node only) but you didn't check anything in this condition.

This will then make every block to be "Node-Only" even when the TW condition is not checked & results in blocks that are not visible on other bundle than node. 

It's a critical issue.